Core/include/Acts/Geometry/detail/DefaultGeometryIdGetter.hpp

```diff
@@ -0,0 +1,45 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Geometry/GeometryID.hpp"
+
+namespace Acts {
+namespace detail {
+
+/// Uniform access to geometry identifiers for different object types.
+struct DefaultGeometryIdGetter {
```

Comment:
Honestly, I would prefer there to be less magic. Can we have the `DefaultGeometryIdGetter` **ONLY** handle the interface that our geometry classes expose, i.e. `geoID()`?

Reply:
This is taken from code that already exists in the framework code. There, all variants are used e.g. to simplify implementing transparent lookup of elements by an equivalent key (compare with [std::set::find](https://en.cppreference.com/w/cpp/container/set/find)). I intended to use this `Getter` there, too.

I could remove everything but the `.geoID()` support for the default variant here; and keep the separate setter in the examples.

Reply:
I would prefer if you did remove that, but I'll leave the decision up to you.

Reply:
I removed everything but the `.geoID()` and `.geometryId()` support.